Lib/fontTools/colorLib/geometry.py

```diff
@@ -0,0 +1,106 @@
+"""Helpers for manipulating 2D points and vectors in COLR table."""
+
+from math import copysign, cos, hypot, pi
+from fontTools.misc.fixedTools import otRound
+
+
+def _vector_between_points(a, b):
+    return (b[0] - a[0], b[1] - a[1])
+
+
+def _distance_between_points(a, b):
+    return hypot(*_vector_between_points(a, b))
+
+
+def _round_point(pt):
+    return (otRound(pt[0]), otRound(pt[1]))
+
+
+def _round_circle(centre, radius):
+    return _round_point(centre), otRound(radius)
+
+
+def _unit_vector(vec):
+    length = hypot(*vec)
+    if length == 0:
+        return None
+    return (vec[0] / length, vec[1] / length)
+
+
+# This is the same tolerance used by Skia's SkTwoPointConicalGradient.cpp to detect
+# when a radial gradient's focal point lies on the end circle.
+_NEARLY_ZERO = 1 / (1 << 12)  # 0.000244140625
+
+
+def _is_circle_inside_circle(c0, r0, c1, r1):
```

Comment:
looks like you have to read the code to know which one is which, consider descriptive names? - c_inner, r_inner, etc might make this more obvious

Reply:
we have a sort of convention elsewhere in the code that c0 is the start circle and c1 is the end circle. It's not really inner or outer.

Reply:
renamed the parameter like you suggested

Reply:
I still find the name/parameters a bit unclear even with that convention. "given a start and an end circle tell me if circle is inside circle" would lead me to ask you to clarify if you mean is start inside end, is end inside start, or if it's fine as long as either one is inside the other.